chart/templates/mirror-public-deployment.yaml

```diff
@@ -0,0 +1,76 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "chart.fullname" . }}-public
+  labels:
+    {{- include "chart.labels" . | nindent 4 }}
+    app: mirror-service-public
+spec:
+  replicas: {{ $.Values.mirrorPublic.replicaCount }}
+  strategy:
+    type: Recreate
+  selector:
+    matchLabels:
+      {{- include "chart.selectorLabels" . | nindent 6 }}
+      app: mirror-service-public
+  template:
+    metadata:
+      annotations:
+        {{- toYaml .Values.mirrorPublic.podAnnotations | nindent 8 }}
+      labels:
+        {{- include "chart.selectorLabels" . | nindent 8 }}
+        app: mirror-service-public
+    spec:
+      imagePullSecrets:
+        {{- toYaml .Values.imagePullSecrets | nindent 8 }}
+      securityContext:
+        runAsUser: 1000
+        runAsGroup: 1000
+      containers:
+      - name: mirror-service-public
+        image: "{{ .Values.mirrorPublic.image.repository }}:{{ .Values.mirrorPublic.image.tag }}"
+        imagePullPolicy: {{ .Values.mirrorPublic.imagePullPolicy }}
+        args:
+          - /usr/local/bin/wallet-service-mirror-public
+          - --client-listen-uri=http://0.0.0.0:9091/
+          {{- if eq .Values.mirrorServiceTLS true }}
+          - --mirror-listen-uri=wallet-service-mirror://0.0.0.0/?tls-chain=/certs/tls.crt&tls-key=/certs/tls.key
```

Comment:
Is it possible to wire in tls.ca-cert or something so you have the whole trust chain? 

Then maybe wouldn't need --allow-self-signed-tls because the ca would be trusted.

Reply:
Looking at the cert-manager issuer it is a self signed type and the secret that is created both the ca.crt and the tls.crt are the same.  I "think" we could deploy a CA issuer that would provide a chain and I found this [Cert-Manager Trust](https://github.com/cert-manager/trust) that we could use to create a configmap in every namespace with the chain.  Something to look at in the future.  
Certs are one of those things that I am sure are easy but my brain struggles with them.  :cry: